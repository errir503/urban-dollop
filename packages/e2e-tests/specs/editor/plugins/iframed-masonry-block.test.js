/**
 * WordPress dependencies
 */
import {
	activatePlugin,
	createNewPost,
	deactivatePlugin,
	insertBlock,
	getEditedPostContent,
	canvas,
	createNewTemplate,
} from '@wordpress/e2e-test-utils';

async function didMasonryLoadCorrectly( context ) {
	return await context.evaluate( () => {
		const container = document.querySelector(
			'.wp-block-test-iframed-masonry-block'
		);
		return (
			// Expect Masonry to set a non-zero height.
			parseInt( container.style.height, 10 ) > 0 &&
			// Expect Masonry to absolute position items.
			container.firstElementChild.style.position === 'absolute'
		);
	} );
}

describe( 'iframed masonry block', () => {
	beforeEach( async () => {
		await activatePlugin( 'gutenberg-test-iframed-masonry-block' );
		await createNewPost( { postType: 'page' } );
	} );

	afterEach( async () => {
		await deactivatePlugin( 'gutenberg-test-iframed-masonry-block' );
	} );

	it( 'should load script and dependencies in iframe', async () => {
		await insertBlock( 'Iframed Masonry Block' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
		expect( await didMasonryLoadCorrectly( canvas() ) ).toBe( true );

		await createNewTemplate( 'Iframed Test' );
		await canvas().waitForSelector( '.grid-item[style]' );

		expect( await didMasonryLoadCorrectly( canvas() ) ).toBe( true );
	} );
} );
