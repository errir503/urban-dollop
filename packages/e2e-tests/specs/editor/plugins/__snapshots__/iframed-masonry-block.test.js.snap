// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iframed masonry block should load script and dependencies in iframe 1`] = `
"<!-- wp:test/iframed-masonry-block -->
<div class="wp-block-test-iframed-masonry-block"><div class="grid-item"></div><div class="grid-item grid-item--width2 grid-item--height2"></div><div class="grid-item grid-item--height3"></div><div class="grid-item grid-item--height2"></div><div class="grid-item grid-item--width3"></div><div class="grid-item"></div><div class="grid-item"></div><div class="grid-item grid-item--height2"></div><div class="grid-item grid-item--width2 grid-item--height3"></div><div class="grid-item"></div><div class="grid-item grid-item--height2"></div><div class="grid-item"></div><div class="grid-item grid-item--width2 grid-item--height2"></div><div class="grid-item grid-item--width2"></div><div class="grid-item"></div><div class="grid-item grid-item--height2"></div><div class="grid-item"></div><div class="grid-item"></div><div class="grid-item grid-item--height3"></div><div class="grid-item grid-item--height2"></div><div class="grid-item"></div><div class="grid-item"></div><div class="grid-item grid-item--height2"></div></div>
<!-- /wp:test/iframed-masonry-block -->"
`;
